pcsx2/IPC.cpp

```diff
@@ -76,26 +76,25 @@ SocketIPC::SocketIPC(SysCoreThread* vm, unsigned int slot)
 	}
 
 #else
+	char* runtime_dir = nullptr;
 #ifdef __APPLE__
-	char* runtime_dir = std::getenv("TMPDIR");
+	runtime_dir = std::getenv("TMPDIR");
 #else
-	char* runtime_dir = std::getenv("XDG_RUNTIME_DIR");
+	runtime_dir = std::getenv("XDG_RUNTIME_DIR");
 #endif
 	// fallback in case macOS or other OSes don't implement the XDG base
 	// spec
-	if (runtime_dir == NULL)
-		m_socket_name = (char*)"/tmp/" IPC_EMULATOR_NAME ".sock";
+	if (runtime_dir == nullptr)
+		m_socket_name = "/tmp/" IPC_EMULATOR_NAME ".sock";
 	else
-		m_socket_name = strcat(runtime_dir, "/" IPC_EMULATOR_NAME ".sock");
-
-	if (slot != IPC_DEFAULT_SLOT)
 	{
-		// maximum size of .%u
-		char slot_ending[34];
-		sprintf(slot_ending, ".%u", slot);
-		m_socket_name = strcat(m_socket_name, slot_ending);
+		m_socket_name = runtime_dir;
+		m_socket_name += "/" IPC_EMULATOR_NAME ".sock";
 	}
 
+	if (slot != IPC_DEFAULT_SLOT)
+		m_socket_name += std::to_string(slot);
```

Comment:
You left out the period before the number here that was in the original code. Intended? (original format string was ".%u")

Reply:
clearly not, as i did it in my client-side commit: https://code.govanify.com/govanify/pcsx2_ipc/commit/029f45e30b3988a3c7acd765f1a69383610aaed5

I swear I tested it though, thanks a lot for the review!